go/exocom/src/message_cache/message_cache.go

```diff
@@ -0,0 +1,59 @@
+package messageCache
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+// MessageCache records the timestamp of each message.
+// The cache automitically deletes any message older then 1 minute
+type MessageCache struct {
+	Cache map[string]time.Time
+	mutex *sync.Mutex
+}
+
+// NewMessageCache returns a new MessageCache
+// removing old messages with a frequency equal to the given duration
+func NewMessageCache(cleanupInterval time.Duration) *MessageCache {
+	result := new(MessageCache)
+	result.mutex = &sync.Mutex{}
+	go func() {
+		for {
+			time.Sleep(cleanupInterval)
+			result.mutex.Lock()
+			result.clearCache()
+			result.mutex.Unlock()
+		}
+	}()
+	return result
+}
+
+func (c *MessageCache) clearCache() {
+	for id, timestamp := range c.Cache {
+		if time.Since(timestamp) > time.Minute {
+			delete(c.Cache, id)
+		}
+	}
+	return
+}
+
+// Get returns the timestamp for the given messageId returning an error if
+// no data is available
+func (c *MessageCache) Get(messageID string) (time.Time, error) {
+	c.mutex.Lock()
+	timestamp, ok := c.Cache[messageID]
+	if ok {
+		c.mutex.Unlock()
```

Comment:
I think you can move this line before the if clause. It is the direct next action in both casese.